File: 프로그래머스/2/42586. 기능개발/기능개발.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        var answer = new List<int>();
        Queue<int> queue = new Queue<int>();
        
        for(int i = 0; i < progresses.Length; i++)
        {
            int remain = 100 - progresses[i];
            int days = (remain + speeds[i] - 1) / speeds[i];
            queue.Enqueue(days);
        }
        int idx = 0;
        while(queue.Count > 0)
        {
            int cur = queue.Dequeue();
            int count = 1;
            while(queue.Count > 0 && queue.Peek() <= cur)
            {
                queue.Dequeue();
                count++;
            }
            answer.Add(count);
        }
        return answer.ToArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> q;
    
    for(int i = 0; i < progresses.size(); i++)
    {
        int day = ceil((100.0 - progresses[i]) / speeds[i]);
        q.push(day);
    }
    
    while(q.size())
    {
        int cur = q.front();
        q.pop();
        
        int cnt = 1;
        
        while(q.size() && q.front() <= cur)
        {
            q.pop();
            cnt++;
        }
        answer.push_back(cnt);
    }
    
    return answer;
}

// 7 3 9